File: LinkedList/SinlgyLinkedList/Java/SLL.java
class SSL<T>{

    private class Node{
        
        T value;
        Node next;

        public Node(T value, Node next){
            this.value = value;
            this.next = next;        
        }
    }

    Node head;

    public SSL(){
        head = null;
    }

    public void insert(T value){        
        Node newNode = new Node(value, head);
        head = newNode;        
    }

    public void remove(){
        head = head.next;
    }

    public boolean contains(T value){
        Node current = head;
        while(current != null){
            if(current.value == value){
                return true;
            }
            current = current.next;
        }
        return false;
    }

    public void printList(){
        Node current = head;
        while(current != null){
            System.out.print(current.value + " ");
            current = current.next;
        }
        System.out.print("\n");
    }
}

File: LinkedList/SinlgyLinkedList/C++/SLL.cpp
#include <iostream>

class Node{

    public:
        int value;
        Node *next;

        Node() = default;
};


class SLL{

    public:
        Node *head;

        SLL(){
            head = nullptr;
        }

        void insert(int value){
            Node *newNode = new Node;
            newNode->value = value;
            newNode->next = head;
            head = newNode;
        }

        void remove(){
            if(head == nullptr){
                return;
            }
            Node *tmp = head;
            head = tmp->next;
            delete tmp;
        }
        
        void printList(){
            Node *current = head;
            while(current != nullptr){
                std::cout << current->value << " ";
                current = current->next;
            }
            std::cout << std::endl;
            free(current);
        }        
};
